modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue

```diff
@@ -9,37 +9,64 @@
+          v-if="field.label"
           class="apos-field__label"
-          :is="labelEl" :for="uid"
+          :class="{'apos-sr-only': field.hideLabel }"
+          :is="labelEl"
+          :for="uid"
+          :data-apos-test-name="field.name"
+          :data-apos-test-label="field.label"
+          data-apos-test="field-label"
         >
-          {{ $t(label) }}
-          <span v-if="field.required" class="apos-field__required">
-            *
-          </span>
-          <AposLabel
-            class="apos-field__tag"
-            v-if="field.tag"
-            :label="field.tag.value || field.tag"
-            :modifiers="[ `apos-is-${field.tag.type || 'success'}`, 'apos-is-filled' ]"
-          />
-          <span
-            v-if="(field.help || field.htmlHelp) && displayOptions.helpTooltip"
-            class="apos-field__help-tooltip"
-          >
-            <AposIndicator
-              icon="help-circle-icon"
-              class="apos-field__help-tooltip__icon"
-              :tooltip="field.help || field.htmlHelp"
-              :icon-size="11"
-              icon-color="var(--a-base-4)"
+          <span class="apos-field_label-info">
+            {{ $t(label) }}
+            <span v-if="field.required" class="apos-field__required">
+              *
+            </span>
+            <AposLabel
+              class="apos-field__tag"
+              v-if="field.tag"
+              :label="field.tag.value || field.tag"
+              :modifiers="[ `apos-is-${field.tag.type || 'success'}`, 'apos-is-filled' ]"
+              data-apos-test="field-tag"
             />
+            <span
+              v-if="(field.help || field.htmlHelp) && displayOptions.helpTooltip"
+              data-apos-test="field-help-tooltip"
+              class="apos-field__help-tooltip"
+            >
+              <AposIndicator
+                icon="help-circle-icon"
+                class="apos-field__help-tooltip__icon"
+                :tooltip="field.help || field.htmlHelp"
+                :icon-size="11"
+                icon-color="var(--a-base-4)"
+              />
+            </span>
+            <span
+              v-if="displayOptions.changed" class="apos-field__changed"
+              data-apos-test="field-changed"
+            >
+              <AposLabel
+                label="apostrophe:changed" class="apos-field__changed__label"
+                :modifiers="[ 'apos-is-warning', 'apos-is-filled' ]"
+                tooltip="apostrophe:fieldHasUnpublishedChanges"
+              />
+            </span>
           </span>
-          <span v-if="displayOptions.changed" class="apos-field__changed">
-            <AposLabel
-              label="apostrophe:changed" class="apos-field__changed__label"
-              :modifiers="[ 'apos-is-warning', 'apos-is-filled' ]"
-              tooltip="apostrophe:fieldHasUnpublishedChanges"
+          <span data-apos-test="field-meta-wrapper" class="apos-field__label-meta">
+            <component
```

Comment:
You're doing this in `AposInputWrapper`... so is there actually any problem with field types not on your "implemented" list or are there just some that don't use the wrapper? If it will automatically work for new field types that use the wrapper that is worth mentioning.

Reply:
Every field has to explicitly pass `:meta="meta"` to the input wrapper in order to let the indicator work. We can do that for the rest of the core fields on the fly. My main concern here is the huge scope that already has this PR.